nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/avro/AvroTypeUtil.java

```diff
@@ -877,6 +878,18 @@ private static Object convertToAvroObject(final Object rawValue, final Schema fi
      */
     private static Object convertUnionFieldValue(final Object originalValue, final Schema fieldSchema, final Function<Schema, Object> conversion, final String fieldName) {
         boolean foundNonNull = false;
+
+        Optional<Schema> mostSuitableType = DataTypeUtils.findMostSuitableType(
+                originalValue,
+                fieldSchema.getTypes().stream().filter(schema -> schema.getType() != Type.NULL).collect(Collectors.toList()),
+                subSchema -> AvroTypeUtil.determineDataType(subSchema)
+        );
+        if (mostSuitableType.isPresent()) {
+            Object convertedVAlue = conversion.apply(mostSuitableType.get());
```

Comment:
This would be made simpler by simply returning `conversion.apply(...)`

Reply:
Yeah, I guess even my previous reasoning doesn't hold well as this method already has multiple return statements. (Although during debugging it still can be useful to just add a break point and see immediately the returned value, some simplification could be better.)